### PR TITLE
[vgg] Added pytorch model for vgg16

### PR DESCRIPTION
**Pull request by kparichay**
Added pytorch model for vgg16
    This is to benchmark against tf and nntrainer
    
Update the nntrainer and tensorflow to use official VGG16 model architecture
    The FC layers setup is different as the cifar100 dataset has just 100 output classes
    than 1000 classes of the imagenet.
    Further, the number of epochs are reduced to 1.
    When training, this can be increased appropriately.

**Self evaluation:**
    1. Build test: [x]Passed [ ]Failed [ ]Skipped
    2. Run test: [x]Passed [ ]Failed [ ]Skipped
    
Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: Applications/VGG/PyTorch/main.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+# Model
+
+cfg = {'VGG16': [64, 64, 'M', 128, 128, 'M', 256, 256, 256, 'M', 512, 512, 512, 'M', 512, 512, 512, 'M']}
+class VGG(nn.Module):
+    def __init__(self, vgg_name):
+        super(VGG, self).__init__()
+        self.features = self._make_layers(cfg[vgg_name])
+        self.classifier = nn.Sequential(*[nn.Linear(512, 256), nn.BatchNorm1d(256),
+                                          nn.ReLU(inplace=True), nn.Linear(256, 100)])
```
Comment (zhoonit) [inline Applications/VGG/PyTorch/main.py:62]: Could you check formatting here?

File: Applications/VGG/PyTorch/main.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    testset, batch_size=128, shuffle=False, num_workers=4)
+
+# Model
+
+cfg = {'VGG16': [64, 64, 'M', 128, 128, 'M', 256, 256, 256, 'M', 512, 512, 512, 'M', 512, 512, 512, 'M']}
+class VGG(nn.Module):
+    def __init__(self, vgg_name):
+        super(VGG, self).__init__()
+        self.features = self._make_layers(cfg[vgg_name])
+        self.classifier = nn.Sequential(*[nn.Linear(512, 256), nn.BatchNorm1d(256),
```
Comment (zhoonit) [inline Applications/VGG/PyTorch/main.py:61]: if * is needed, I guess wrapping in to a list is can go away with *.

`nn.Seqeuntial(nn.Linear, Batchnorm...)`

File: Applications/VGG/PyTorch/main.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        'net': net.state_dict(),
+        'acc': acc,
+        'epoch': epoch,
+    }
+    if not os.path.isdir('checkpoint'):
+        os.mkdir('checkpoint')
+    torch.save(state, './checkpoint/ckpt.pth')
+
+
+for epoch in range(1):
```
Comment (zhoonit) [inline Applications/VGG/PyTorch/main.py:148]: Could you add if __name__ == '__main__' to state this is start of a script?

**Review by zhoonit — APPROVED** [review]
LGTM :)

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: Applications/VGG/PyTorch/main.py (hunk 1 above)
Comment (kparichay) [reply Applications/VGG/PyTorch/main.py]: done.

File: Applications/VGG/PyTorch/main.py (hunk 2 above)
Comment (kparichay) [reply Applications/VGG/PyTorch/main.py]: done.

**Comment by kparichay** [comment]
With this patch, below is the peak heap memory consumption (calculated using HeapTrack) of VGG16 model on CIFAR100 data with CPU backend on ubuntu desktop 16.04 (python version 3.5.2):

**PyTorch - version 1.5.1, torchvision - version 0.2.2**
calls to allocation functions: 12890705 (4851/s)
temporary memory allocations: 2484467 (935/s)
peak heap memory consumption: 983.68MB
peak RSS (including heaptrack overhead): 0B
total memory leaked: 50.51MB

**Tensorflow - vesion 2.2.1**
calls to allocation functions: 31701691 (36360/s)
temporary memory allocations: 2112413 (2422/s)
peak heap memory consumption: 782.88MB
peak RSS (including heaptrack overhead): 0B
total memory leaked: 25.36MB

**NNTrainer - master branch with commit 5e253e635b67ab9ba5641956ec336ef98b168b7b**
calls to allocation functions: 1629537 (1192/s)
temporary memory allocations: 13201 (9/s)
peak heap memory consumption: 519.11MB
peak RSS (including heaptrack overhead): 0B
total memory leaked: 136.14KB
